upload.sh

```diff
@@ -1,6 +1,7 @@
 #!/bin/bash
 
-IFS=';' read -ra DIRECTORIES <<<"${COMPONENTS_DIRECTORIES:-.}"
+COMPONENTS_DIRECTORIES=$(echo -e $COMPONENTS_DIRECTORIES | tr -d '[:space:]')
+IFS=';' read -a DIRECTORIES <<<"${COMPONENTS_DIRECTORIES:-.}"
```

Comment:
I suggest wrapping `"$COMPONENTS_DIRECTORIES"` in double quotes to work correctly on an empty string. You can also use put default substitution inline.



```suggestion
IFS=';' read -ra DIRECTORIES <<<"$(echo -e "${COMPONENTS_DIRECTORIES:-.}" | tr -d '[:space:]')"
```

Reply:
This is a well-grounded suggestion and I agree it is more suitable than my solution.